packages/asset-swapper/src/utils/aggregate.ts

```diff
@@ -0,0 +1,800 @@
+import { IERC20BridgeSamplerContract } from '@0x/contracts-erc20-bridge-sampler';
+import { generatePseudoRandomSalt } from '@0x/order-utils';
+import { SignedOrderWithoutDomain } from '@0x/types';
+import { AbiEncoder, BigNumber } from '@0x/utils';
+
+import { constants } from '../constants';
+
+const { NULL_BYTES, NULL_ADDRESS } = constants;
+const ZERO = new BigNumber(0);
+const INFINITE_TIMESTAMP_SEC = new BigNumber(2524604400);
+// TODO(dorothy-zbornak): Point these addresses to the mainnet.
+export const KYBER_BRIDGE_ADDRESS = '0xa0cdca934847556eaf431d909fb3cf4aca01df66';
+export const ETH2DAI_BRIDGE_ADDRESS = '0x27b8c4473c6b885d0b060d722cf614dbc3f9adfb';
+export const UNISWAP_BRIDGE_ADDRESS = '0x9b81c8beee5d0ff8b128adc04db71f33022c5163';
+
+/**
+ * Common exception messages thrown by aggregation logic.
+ */
+export enum AggregationError {
+    NoOptimalPath = 'no optimal path',
+    EmptyOrders = 'empty orders',
+}
+
+/**
+ * DEX sources to aggregate.
+ */
+export enum ERC20BridgeSource {
+    Native,
+    Uniswap,
+    Eth2Dai,
+    Kyber,
+    NumSources,
+}
+
+/**
+ * Convert a source to a canonical address used by the sampler contract.
+ */
+export const SOURCE_TO_ADDRESS: { [key: string]: string } = {
+    [ERC20BridgeSource.Eth2Dai]: '0x39755357759cE0d7f32dC8dC45414CCa409AE24e',
+    [ERC20BridgeSource.Uniswap]: '0xc0a47dFe034B400B47bDaD5FecDa2621de6c4d95',
+    [ERC20BridgeSource.Kyber]: '0x818E6FECD516Ecc3849DAf6845e3EC868087B755',
+};
+
+/**
+ * Valid sources for market sell.
+ */
+export const SELL_SOURCES = [ERC20BridgeSource.Uniswap, ERC20BridgeSource.Eth2Dai, ERC20BridgeSource.Kyber];
+
+/**
+ * Valid sources for market buy.
+ */
+export const BUY_SOURCES = [ERC20BridgeSource.Uniswap, ERC20BridgeSource.Eth2Dai];
+
+/**
+ * Represents a node on a fill path.
+ */
+export interface Fill {
+    // See `FillFlags`.
+    flags: number;
+    // `FillFlags` that are incompatible with this fill, e.g., to prevent
+    // Kyber from mixing with Uniswap and Eth2Dai and vice versa.
+    exclusionMask: number;
+    // Input fill amount (taker asset amount in a sell, maker asset amount in a buy).
+    input: BigNumber;
+    // Output fill amount (maker asset amount in a sell, taker asset amount in a buy).
+    output: BigNumber;
+    // Fill that must precede this one. This enforces certain fills to be contiguous.
+    parent?: Fill;
+    // Arbitrary data to store in this Fill object. Used to reconstruct orders
+    // from paths.
+    data?: any;
```

Comment:
Feel like a more verbose name here other than data would help

Reply:
This should actually just be `FillData`, if we need more data entries, we can always add it.

We can use a union type to capture all of the 'types' that `data` can be.